Pocket/pocket-ios
# Alternate Icons selection

## Summary
* This PR allows users to select a custom icon among the available ones, that can be found in Settings -> App Icon

## References 
* Branch name

## Implementation Details
* Add `SelectIconView`, a `SwiftUI` view that lets users select an icon among the available ones
* Add `SelectIconViewModel`, view model of the aforementioned view, responsible to set the selected icon
* Add `PocketAppIcon`, a type that describes an icon

> [!NOTE]
> The used icon is temporary and while we wait for other icons, we can use this one to try the feature out

## Test Steps
* Build/run then go to Settings and tap "App Icon"
* Select an icon that is not currently selected and verify that it's set as the app icon (see animation below)

## PR Checklist:

## Screenshots
<p align=center>
<img height=700 src=https://github.com/Pocket/pocket-ios/assets/34376330/68e93d8b-7994-4107-9521-0956618d996e>
</p>
